reframe/core/modules.py

```diff
@@ -203,6 +204,27 @@ def is_module_loaded(self, name):
     def _is_module_loaded(self, name):
         return self._backend.is_module_loaded(Module(name))
 
+    def load_mapping(self, mapping):
+        key, values = mapping.strip().strip(':').split(':')
+        self.module_map[key.strip()] = list(
+            OrderedDict.fromkeys(values.split()))
+
+    def load_mapping_from_file(self, filename):
+        """Update the internal module mapping from a file."""
+
+        with open(filename, 'r') as fp:
+            for lineno, line in enumerate(fp):
+                line = line.strip().split('#')[0]
+                if not line:
+                    continue
+
+                try:
+                    self.load_mapping(line)
+                except Exception as e:
+                    raise ConfigError('Line %d of "%s": '
+                                      'Invalid format of the module mapping'
+                                      'file.\n %s' % (lineno, filename, e))
```

Comment:
The proper way of reraising this is the following:

```python
try:
    may_throw_op()
except ConfigError as e:
    raise ConfigError('%s:%s' % (filename, lineno)) from e
```

This will print out `/path/to/map/file:55: message from e`.